servers/rendering/renderer_rd/renderer_scene_render_rd.cpp

```diff
@@ -422,7 +422,7 @@ void RendererSceneRenderRD::_render_buffers_post_process_and_tonemap(const Rende
 	// Glow, auto exposure and DoF (if enabled).
 
 	Size2i target_size = rb->get_target_size();
-	bool can_use_effects = target_size.x >= 8 && target_size.y >= 8; // FIXME I think this should check internal size, we do all our post processing at this size...
+	bool can_use_effects = target_size.x >= 8 && target_size.y >= 8 && debug_draw == RS::VIEWPORT_DEBUG_DRAW_DISABLED; // FIXME I think this should check internal size, we do all our post processing at this size...
```

Comment:
Hi @clayjohn,

> This is going to disable post processing for all debug draw modes including draw modes like "Shadow atlas" which just adds an overlay in the corner of the screen. I think users would be surprised that drawing the shadow atlas disables post processing, especially if they use auto-exposure.

I agree with you that it may be misleading to users.

> We should take a look at all the debug draw modes and figure out which ones actually need to disable post processing effects.

I will try to look at all draw modes (based on ViewportDebugDraw enum) and see for which ones it is relevant to use or not post effects. But I may not be the better one for this task, I will tell you if I am lost.

> Also note, many debug draw modes are implemented as a full screen quad drawn over top of the screen. So for those ones it doesn't matter whether post effects are disabled or not.

It will be less unneeded computation if we disable post effects in this case, no ? Or it will be insignificant ?

Reply:
> It will be less unneeded computation if we disable post effects in this case, no ? Or it will be insignificant ?

It would be slightly less computation, but it doesn't matter.

I did a pass over the debug modes and wrote down whether I thought they should be enabled or not

```
VIEWPORT_DEBUG_DRAW_DISABLED, // Enabled
VIEWPORT_DEBUG_DRAW_UNSHADED, // Disabled
VIEWPORT_DEBUG_DRAW_LIGHTING, // Enabled
VIEWPORT_DEBUG_DRAW_OVERDRAW, // Disabled
VIEWPORT_DEBUG_DRAW_WIREFRAME, // Disabled
VIEWPORT_DEBUG_DRAW_NORMAL_BUFFER, // Draws over screen, doesn't matter
VIEWPORT_DEBUG_DRAW_VOXEL_GI_ALBEDO, // Disabled
VIEWPORT_DEBUG_DRAW_VOXEL_GI_LIGHTING, // Enabled 
VIEWPORT_DEBUG_DRAW_VOXEL_GI_EMISSION, // Enabled
VIEWPORT_DEBUG_DRAW_SHADOW_ATLAS, // Enabled
VIEWPORT_DEBUG_DRAW_DIRECTIONAL_SHADOW_ATLAS, // Enabled
VIEWPORT_DEBUG_DRAW_SCENE_LUMINANCE, // Enabled
VIEWPORT_DEBUG_DRAW_SSAO, // Draws over screen, doesn't matter
VIEWPORT_DEBUG_DRAW_SSIL, // Draws over screen, doesn't matter
VIEWPORT_DEBUG_DRAW_PSSM_SPLITS, // Enabled
VIEWPORT_DEBUG_DRAW_DECAL_ATLAS, // Enabled
VIEWPORT_DEBUG_DRAW_SDFGI, // Disabled
VIEWPORT_DEBUG_DRAW_SDFGI_PROBES, // Enabled
VIEWPORT_DEBUG_DRAW_GI_BUFFER, // Draws over screen, doesn't matter
VIEWPORT_DEBUG_DRAW_DISABLE_LOD, // Enabled
VIEWPORT_DEBUG_DRAW_CLUSTER_OMNI_LIGHTS, // Draws over screen, doesn't matter
VIEWPORT_DEBUG_DRAW_CLUSTER_SPOT_LIGHTS, // Draws over screen, doesn't matter
VIEWPORT_DEBUG_DRAW_CLUSTER_DECALS, // Draws over screen, doesn't matter
VIEWPORT_DEBUG_DRAW_CLUSTER_REFLECTION_PROBES,
VIEWPORT_DEBUG_DRAW_OCCLUDERS, // Draws over screen, doesn't matter
VIEWPORT_DEBUG_DRAW_MOTION_VECTORS, // Enabled
VIEWPORT_DEBUG_DRAW_INTERNAL_BUFFER, // Draws over screen, doesn't matter
```

Reply:
On my side, I tested every mode with the DOF blur.
I used godot docs https://docs.godotengine.org/en/stable/classes/class_viewport.html#enum-viewport-debugdraw to understand how to use each mode.

And I have some small differences with you in my choices:

> VIEWPORT_DEBUG_DRAW_VOXEL_GI_LIGHTING, // Enabled 
> VIEWPORT_DEBUG_DRAW_VOXEL_GI_EMISSION, // Enabled
> VIEWPORT_DEBUG_DRAW_PSSM_SPLITS, // Enabled

I think it should be disable in this cases, but its just a matter of taste on my side.

> VIEWPORT_DEBUG_DRAW_SCENE_LUMINANCE, // Enabled

I only see the DEBUG_DRAW_DISABLED mode.
I don’t know what it should show. Godot docs is mute on this enum value.

> VIEWPORT_DEBUG_DRAW_SSAO, // Draws over screen, doesn't matter
> VIEWPORT_DEBUG_DRAW_SSIL, // Draws over screen, doesn't matter
> VIEWPORT_DEBUG_DRAW_GI_BUFFER, // Draws over screen, doesn't matter
> VIEWPORT_DEBUG_DRAW_OCCLUDERS, // Draws over screen, doesn't matter

Drawn over viewport but when the texture is not available it will show as DEBUG_DRAW_DISABLED.
So either we use camera effects on those modes, or we should check that the corresponding texture is available before disabling them.

> VIEWPORT_DEBUG_DRAW_SDFGI_PROBES, // Enabled

Camera effects affects the probes… I don’t think we want that. I propose to disable them.

> VIEWPORT_DEBUG_DRAW_CLUSTER_OMNI_LIGHTS, // Draws over screen, doesn't matter
> VIEWPORT_DEBUG_DRAW_CLUSTER_SPOT_LIGHTS, // Draws over screen, doesn't matter
> VIEWPORT_DEBUG_DRAW_CLUSTER_DECALS, // Draws over screen, doesn't matter
> VIEWPORT_DEBUG_DRAW_CLUSTER_REFLECTION_PROBES,
> VIEWPORT_DEBUG_DRAW_INTERNAL_BUFFER, // Draws over screen, doesn't matter

Camera effects are affecting rendering of those modes, so I didn't mark them as "drawn over screen" in my notes. I propose to disable camera effects on them.

Reply:
To sum up, I propose to disable camera effects on following modes:
```
VIEWPORT_DEBUG_DRAW_UNSHADED
VIEWPORT_DEBUG_DRAW_OVERDRAW
VIEWPORT_DEBUG_DRAW_WIREFRAME
VIEWPORT_DEBUG_DRAW_NORMAL_BUFFER
VIEWPORT_DEBUG_DRAW_VOXEL_GI_ALBEDO 
VIEWPORT_DEBUG_DRAW_VOXEL_GI_LIGHTING
VIEWPORT_DEBUG_DRAW_VOXEL_GI_EMISSION
VIEWPORT_DEBUG_DRAW_PSSM_SPLITS
VIEWPORT_DEBUG_DRAW_SDFGI
VIEWPORT_DEBUG_DRAW_SDFGI_PROBES
VIEWPORT_DEBUG_DRAW_GI_BUFFER
VIEWPORT_DEBUG_DRAW_CLUSTER_OMNI_LIGHTS
VIEWPORT_DEBUG_DRAW_CLUSTER_SPOT_LIGHTS
VIEWPORT_DEBUG_DRAW_CLUSTER_DECALS
VIEWPORT_DEBUG_DRAW_CLUSTER_REFLECTION_PROBES
VIEWPORT_DEBUG_DRAW_INTERNAL_BUFFER
```

Reply:
This list looks good to me, however, `VIEWPORT_DEBUG_DRAW_SDFGI` should have post-processing disabled as I believe it draws cascades over the screen. Probes on the other hand could leave post-processing enabled.

Reply:
`VIEWPORT_DEBUG_DRAW_SDFGI` is not affected by DOF blur but when there is no SDGFI and no VoxelGI active on your project the rendering is the `VEWPORT_DEBUG_DRAW_DISABLED` and camera effects should be enabled.
It seems to me that it is a "drawn over viewport" case. And if it is the case, as I said in my long detailed message, I think that in case of the buffer (drawn over viewport) is not available we should enable camera effects for the normal rendering behind.

`VIEWPORT_DEBUG_DRAW_SDFGI_PROBES` draws 3D probes in the screen and they are affected by DOF blur, do we really want that ?

Reply:
@Calinou @clayjohn
I updated the PR, tell me if we agree on the modes that should disable camera effects.

Reply:
The list of disabled modes looks good to me now. I tested the latest revision of this PR locally and depth of field disabling worked as expected.

Reply:
@clayjohn I tried to make a mix of your list and my out of experimentation list...
You have something to add ? can it be merged now ?